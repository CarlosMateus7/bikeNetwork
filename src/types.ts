export interface Location {
    latitude: number;
    longitude: number;
    city: string;
    country: string;
}

export interface Network {
    company:string[];
    href:string;
    location: Location;
    name:string;
    id:string;
}

export interface Networks{
    networks:Network[];
}

export interface NetworkError{
    error: string;
}

export interface Station {
    name: string;
    timestamp: string;
    longitude: number;
    latitude: number;
    free_bikes: number;
    empty_slots: number;
    id: string;
}

export interface NetworkById {
    name: string;
    stations: Station[];
    company: string[];
    href: string;
    location: Location;
    id: string;
}