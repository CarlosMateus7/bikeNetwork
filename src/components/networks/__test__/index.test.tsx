import { render, fireEvent, screen } from '@testing-library/react';
import { Networks } from '../index';
import { Network } from '../../../types';

describe('Networks component', () => {
  const network: Network = {
    company: ['EMEL'],
    href: "site",
    location: {
      country: 'PT',
      city: 'Lisbon',
      latitude: 38.724954,
      longitude: -9.149334,
    },
    name: "Gira",
    id:"gira"
  };

  const numberOfNetworksPerCountry = 5;

  it('should make a Snapshot of the Networks component',()=>{
    const { asFragment } = render(
        <Networks network={network} numberOfNetworksPerCountry={numberOfNetworksPerCountry} />
      );
      expect(asFragment()).toMatchSnapshot();
    });

  it('should render L1 number of networks', () => {
    render(
      <Networks network={network} numberOfNetworksPerCountry={numberOfNetworksPerCountry} />
    );
    
    const renderOfNetworkText=screen.getByText(`${numberOfNetworksPerCountry}`);

    expect(renderOfNetworkText).toBeInTheDocument();
  });

  it('should render info for L2 Number of Stations text', () => {
    render(
      <Networks network={network} numberOfNetworksPerCountry={numberOfNetworksPerCountry} />
    );
  
    fireEvent.click(screen.getByText('L2'));
    const numberOfStations = screen.getByText(`Number of Stations (network):`)

    expect(numberOfStations).toBeInTheDocument();
  });

  it('should render info for L3 Country and City', () => {
    render(
      <Networks network={network} numberOfNetworksPerCountry={numberOfNetworksPerCountry} />
    );

    fireEvent.click(screen.getByText('L3'));

    expect(screen.getByText(`${network.location.country}`)).toBeInTheDocument();
    expect(screen.getByText(`${network.location.city}`)).toBeInTheDocument();
  });
});
