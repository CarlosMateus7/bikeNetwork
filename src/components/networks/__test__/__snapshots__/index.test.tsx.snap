// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Networks component should make a Snapshot of the Networks component 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Network: Gira
    </h3>
    <nav
      class="nav-layer"
    >
      <ul
        class="ul-layer"
      >
        <li
          class="li-layer active"
        >
          L1
        </li>
        <li
          class="li-layer "
        >
          L2
        </li>
        <li
          class="li-layer "
        >
          L3
        </li>
      </ul>
    </nav>
    <div>
      <div>
        <div>
          <p
            class="li-layer"
          >
            <strong>
              Number of Networks (PT): 
            </strong>
            5
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
